cdap-metrics-writer-ext-gcp-monitoring/src/main/java/io/cdap/cdap/metrics/process/gcp/CloudMonitoringWriter.java

```diff
@@ -137,6 +138,20 @@ public void initialize(MetricsWriterContext metricsWriterContext) {
   }
 
   private void populateAutoFilledMap(Map<String, String> properties, String platformVersion) {
+    if (platformVersion == null || platformVersion.isEmpty()) {
+      throw new IllegalArgumentException("Missing value for platformVersion");
+    }
+    String[] requiredProperties = {PROJECT, ORG_ID, LOCATION, CLUSTER_ID, INSTANCE_ID};
+    for (String requiredProperty : requiredProperties) {
+      if (!properties.containsKey(requiredProperty)) {
+        throw new IllegalArgumentException("Missing value for " + requiredProperty);
+      }
+      String propertyValue = properties.get(requiredProperty);
+      if (propertyValue == null || propertyValue.isEmpty()) {
+        throw new IllegalArgumentException("Missing value for " + requiredProperty);
```

Comment:
nit: Please make this error message slightly different so we can distinguish between the config being an empty string/null and the config not being there. Thanks!

Reply:
Done